src/Features/Core/Portable/ExtractInterface/AbstractExtractInterfaceService.cs

```diff
@@ -298,25 +397,43 @@ private static Solution GetSolutionWithFormattedInterfaceDocument(Document unfor
             // extracted members are also not updated.
             if (typeToExtractFrom.TypeKind == TypeKind.Interface)
             {
-                return solutionWithFormattedInterfaceDocument;
+                return solution;
             }
 
-            var formattedSolution = GetSolutionWithUpdatedOriginalType(
-                solutionWithFormattedInterfaceDocument,
+            var unformattedSolution = solution;
+            foreach (var documentId in documentIds)
+            {
+                var document = solution.GetDocument(documentId);
+                var currentRoot = await document.GetSyntaxRootAsync().ConfigureAwait(false);
+                var editor = new SyntaxEditor(currentRoot, solution.Workspace);
+
+                var syntaxGenerator = SyntaxGenerator.GetGenerator(document);
+                var typeReference = syntaxGenerator.TypeExpression(extractedInterfaceSymbol);
+
+                foreach (var typeDeclaration in currentRoot.GetAnnotatedNodes(typeNodeAnnotation))
```

Comment:
when would you have multiple annotated nodes?

Reply:
Good point, I'll go to single